aiohttp/connector.py

```diff
@@ -490,8 +495,18 @@ async def connect(
         """Get from pool or create new connection."""
         key = req.connection_key
         available = self._available_connections(key)
-        wait_for_conn = available <= 0 or key in self._waiters
-        if not wait_for_conn and (proto := self._get(key)) is not None:
+        fut: Optional[asyncio.Future[None]] = None
+        waiters: Optional[Deque[asyncio.Future[None]]] = None
+        if wait_for_conn := available <= 0 or key in self._waiters:
```

Comment:
I think `key in self._waiters` might still be a race risk because one of the waiters could be cancelled